server/filestore.go

```diff
@@ -2198,7 +2203,11 @@ func (fs *fileStore) expireMsgsOnRecover() {
 			// Update fss
 			// Make sure we have fss loaded.
 			mb.removeSeqPerSubject(sm.subj, seq)
-			fs.removePerSubject(sm.subj)
+			if fs.removePerSubject(sm.subj) {
+				mb.mu.Unlock()
```

Comment:
A bit nervous here releasing and re-acquiring the mb lock. Also we do not need to do this unless the stream is configured to want tombstones yes? Might be good to avoid this all together if not configured for it.

Reply:
If we don't release the lock then `storeRawMsg` deadlocks if it is writing the tombstone into the same block, so not sure what the best way around that is at the moment. At this point though the overall filestore lock is held so not sure it matters.